lib/house.rb

```diff
@@ -0,0 +1,36 @@
+class House
+
+  attr_reader :price, :address, :rooms
+
+  def initialize(price, address)
+    @price = price
+    @address = address
+    @rooms = []
+  end
+
+  def add_room(room)
+    @rooms << room
+  end
+
+  def above_market_average?
+    house_price = @price.chars.map { |char| char.to_i }
+    house_price.join.to_i > 500000
+  end
+
+  def rooms_from_category(category)
+    rooms_from_category = @rooms.find_all { |room| room.category == category }
```

Comment:
Great use of find_all, but you don't really need to assign the result to a variable - you can let the return value of the find_all be the return value of the method.